src/LayerPlanBuffer.h

```diff
@@ -27,12 +27,14 @@ class LayerPlanBuffer : SettingsMessenger
 
     static constexpr const double extra_preheat_time = 1.0; //!< Time to start heating earlier than computed to avoid accummulative discrepancy between actual heating times and computed ones.
 
+    std::vector<bool> extruder_used_in_meshgroup; //!< For each extruder whether it has already been planned once in this meshgroup. This is used to see whether we should heat to the initial_print_temp or to the printing_temperature
```

Comment:
We now have:
* `ExtruderTrain::getIsUsed`
* `GCodeExport::getExtruderIsUsed`
* `LayerPlanBuffer::extruder_used_in_meshgroup`
* `SliceDataStorage::getExtrudersUsed`